### PR TITLE
Test coverage for selected XSS protections

### PR DESCRIPTION
**Pull request by eliotsykes**

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -18,14 +18,17 @@ class ApplicationMailer < ActionMailer::Base
...
-    (template_string % context).html_safe
+    render_with_html_escaping(template_string, context)
+  end
+
+  private
+
+  def render_with_html_escaping(template_string, context)
+    auto_escaping_template_string = ActiveSupport::SafeBuffer.new(template_string)
+    # SafeBuffer overrides % operator to auto-escape values
+    auto_escaping_template_string % context
```
Comment (eliotsykes) [inline app/mailers/application_mailer.rb]: @mhgoldman just wanted to highlight a minor change to `render_setting_as_html`. This no longer scans for keys ending with `_table` and doesn't escape them. Instead it relies on the built-in behaviour of ActiveSupport::SafeBuffer (subclasses String). 

When using template strings, `ActiveSupport::SafeBuffer` will HTML escape all values, unless they've been marked as html_safe (the *_table values are already automatically marked as html_safe as they were generated with `capture do ... end` in an html.erb template). 

So in other words, pretty much the same end result: all values are html escaped except for the table HTML values.
